spk/owncloud/src/service-setup.sh

```diff
@@ -0,0 +1,507 @@
+
+# ownCloud service setup
+WEB_DIR="/var/services/web_packages"
+# for backwards compatability
+if [ $SYNOPKG_DSM_VERSION_MAJOR -lt 7 ];then
+    WEB_DIR="/var/services/web"
+fi
+if [ -z "${SYNOPKG_PKGTMP}" ]; then
+    SYNOPKG_PKGTMP="${SYNOPKG_PKGDEST_VOL}/@tmp"
+fi
+
+# Others
+OCROOT="${WEB_DIR}/${SYNOPKG_PKGNAME}"
+SQLITE="/bin/sqlite3"
+JQ="/bin/jq"
+SED="/bin/sed"
+SYNOSVC="/usr/syno/sbin/synoservice"
+
+if [ ${SYNOPKG_DSM_VERSION_MAJOR} -lt 7 ]; then
+    GROUP="http"
+fi
+
+set_owncloud_permissions ()
+{
+    if [ ${SYNOPKG_DSM_VERSION_MAJOR} -lt 7 ]; then
+        DIRAPP=$1
+        DIRDATA=$2
+        echo "Setting the correct ownership and permissions of the files and folders in ${DIRAPP}"
+        # Set the ownership for all files and folders to sc-owncloud:http
+        find -L ${DIRAPP} -type d -print0 | xargs -0 chown ${EFF_USER}:${GROUP} 2>/dev/null
+        find -L ${DIRAPP} -type f -print0 | xargs -0 chown ${EFF_USER}:${GROUP} 2>/dev/null
+        # Use chmod on files and directories with different permissions
+        # For all files use 0640
+        find -L ${DIRAPP} -type f -print0 | xargs -0 chmod 640 2>/dev/null
+        # For all directories use 0750
+        find -L ${DIRAPP} -type d -print0 | xargs -0 chmod 750 2>/dev/null
+        # For external data directory
+        if [ -n "${DIRDATA}" ] && [ -d "${DIRDATA}" ]; then
+            chown -R ${EFF_USER}:${GROUP} ${DIRDATA} 2>/dev/null
+            find -L ${DIRDATA} -type f -print0 | xargs -0 chmod 640 2>/dev/null
+            find -L ${DIRDATA} -type d -print0 | xargs -0 chmod 750 2>/dev/null
+        fi
+        # Set the occ command to executable
+        chmod +x ${DIRAPP}/occ 2>/dev/null
+    else
+        echo "Notice: set_owncloud_permissions() is no longer required on DSM7."
+    fi
+}
+
+exec_occ() {
+    PHP="/usr/local/bin/php74"
+    OCC="${OCROOT}/occ"
+    OCC_ARGS=()
+    for arg in "$@"; do
+        OCC_ARGS+=("$arg")
+    done
+    COMMAND="${PHP} ${OCC} ${OCC_ARGS[@]}"
```

Comment:
Just curious, what's the difference with the below

```suggestion
    COMMAND="${PHP} ${OCC} $*"
```

Reply:
I was not familiar with that syntax actually. This is certainly a lot simpler and I can look to implement and test.